controllers/image_generator.go

```diff
@@ -259,11 +271,21 @@ func (r *ImageGenerator) getPeerPodsCM() (*corev1.ConfigMap, error) {
 	if err != nil {
 		return nil, err
 	}
-	// TODO: check in peer-pods-secret as well?
-
 	return peerPodsCM, nil
 }
 
+func (r *ImageGenerator) getPeerPodsSecret() (*corev1.Secret, error) {
+	peerPodsSecret := &corev1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Name:      peerPodsSecretName,
+		Namespace: "openshift-sandboxed-containers-operator",
+	}, peerPodsSecret)
+	if err != nil {
+		return nil, err
+	}
+	return peerPodsSecret, nil
+}
+
```

Comment:
This is only called by the validate function. How are these secrets actually used ?

Reply:
This secret is being mounted along with the CAA pods, the image creation Jobs and fetched by the peerpod-ctrl, here this function was created only for consistency with the getPeerPodCM function.